framework/src/org/checkerframework/framework/test/TestConfigurationBuilder.java

```diff
@@ -0,0 +1,363 @@
+package org.checkerframework.framework.test;
+
+import org.checkerframework.framework.util.PluginUtil;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class TestConfigurationBuilder {
+
+
+    private List<File> diagnosticFiles;
+    private List<File> testSourceFiles;
+    private Set<String> processors;
+    private SimpleOptionMap options;
+
+    private boolean shouldEmitDebugInfo;
+
+    public TestConfigurationBuilder() {
```

Comment:
Should this constructor be private to force usage of the static factory methods?
Should those factory methods be moved to the top of the file to make them easier to find?